### DISCUSSION TITLE
Why do observations and Procedure not occur as often as they should?

### DISCUSSION THREAD

**Discussion by MikaKat** [synthetichealth/synthea#1128] [Q&A] [answered]
Hey everyone,

I want to build a module for laryngeal cancer in Synthea for my PhD thesis. I started testing Synthea with building small and easy modules to understand  the functionality. But I stumbled across a problem where i am not sure if i am using Synthea wrong, its the intented way of working or a bug. When creating a linear module (only transitions of the "direct" typ), then every created Patient should get every conditon, observation and Procedure - right? 

But during my tests I realised that while every created patient gets the conditon, only roughly 10% get the procedure and observation. 

In my understanding 100% of the patients should get the procedures/ observations. What am I doing wrong?

I copyed the module into the synthea modules and use the following command:

`./run_synthea -p 200`

This is my module: 
`{
  "name": "Test-Module",
  "remarks": [
    "A blank module"
  ],
  "states": {
    "Initial": {
      "type": "Initial",
      "name": "Initial",
      "direct_transition": "Routine_Colonoscopy_Encounter"
    },
    "Terminal": {
      "type": "Terminal",
      "name": "Terminal"
    },
    "Detect_Adenoma": {
      "type": "ConditionOnset",
      "target_encounter": "Routine_Colonoscopy_Encounter",
      "assign_to_attribute": "colorectal_adenoma",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": "68496003",
          "display": "Polyp of colon",
          "value_set": ""
        }
      ],
      "name": "Detect_Adenoma",
      "direct_transition": "Fecal_Test_Negative_Result"
    },
    "Routine_Colonoscopy_Encounter": {
      "type": "Encounter",
      "encounter_class": "ambulatory",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": "185349003",
          "display": "Encounter for 'check-up'",
          "value_set": "100%"
        }
      ],
      "name": "Routine_Colonoscopy_Encounter",
      "direct_transition": "Detect_Adenoma",
      "telemedicine_possibility": "always"
    },
    "Routine_Colonoscopy_Procedure": {
      "type": "Procedure",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": "73761001",
          "display": "Colonoscopy",
          "value_set": ""
        }
      ],
      "duration": {
        "low": 25,
        "high": 45,
        "unit": "minutes"
      },
      "name": "Routine_Colonoscopy_Procedure",
      "direct_transition": "End_Routine_Colonoscopy_Encounter",
      "reason": "Detect_Adenoma"
    },
    "End_Routine_Colonoscopy_Encounter": {
      "type": "EncounterEnd",
      "name": "End_Routine_Colonoscopy_Encounter",
      "direct_transition": "Terminal"
    },
    "Fecal_Test_Negative_Result": {
      "type": "Observation",
      "category": "laboratory",
      "range": {
        "low": 0,
        "high": 25
      },
      "unit": "ng/mL",
      "codes": [
        {
          "system": "LOINC",
          "code": "57905-2",
          "display": "Hemoglobin.gastrointestinal",
          "value_set": ""
        }
      ],
      "name": "Fecal_Test_Negative_Result",
      "direct_transition": "Routine_Colonoscopy_Procedure"
    }
  },
  "gmf_version": 2
}`

Already thank you for your help!

**Comment by eedrummer** [comment, marked as answer]
My suggestion would be to add a Guard state to the start of your module.

What I think it happening is that your module is essentially executing at birth. Then, when Synthea exports data, by default, it will truncate to the last 10 years of history. It keeps things like conditions, but you're likely losing the procedure and observation.

The remaining ones where you do see the procedures and observations are likely from the default Colorectal Cancer module.

I would add a guard for age to see if that helps.

**Reply by jawalonoski** [reply]
This is very likely the answer to your problem. You're giving colonoscopies to newborn babies, and as they grow up, that fact is filtered out of the health record (you can change the history kept with the `exporter.years_of_history` parameter.

```properties
# number of years of history to keep in exported records, anything older than this may be filtered out
# set years_of_history = 0 to skip filtering altogether and keep the entire history
exporter.years_of_history = 10
```

**Reply by MikaKat** [reply]
Thank you guys for the quick reply! This was my problem.
Changing to:  `exporter.years_of_history = 0` worked. 

Adding a guard state didn't work, but i think i understand now why. In ca. 90%  of the patients the observation/ procedures is longer then 10 years ago or they didnt turn 30 years (in my case the random age i choose for the guard) yet.
